### PR TITLE
Fix 'Segmentation fault on String.Replace'

### PR DESCRIPTION
**Pull request by LaszloLango**
Related issue: #747

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-regexp-object.cpp [hunk 1]
```diff
@@ -1261,20 +1261,21 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
 
   MEM_DEFINE_LOCAL_ARRAY (input_buffer_p, input_string_size, lit_utf8_byte_t);
 
+  re_matcher_ctx_t re_ctx;
   ssize_t sz = ecma_string_to_utf8_string (input_string_p, input_buffer_p, (ssize_t) input_string_size);
   JERRY_ASSERT (sz >= 0);
 
   lit_utf8_byte_t *input_curr_p = input_buffer_p;
+  re_ctx.input_start_p = input_buffer_p;
```
Comment (akiss77) [inline jerry-core/ecma/operations/ecma-regexp-object.cpp]: if I'd like to be picky -- which I'm obviously not :) -- then I'd suggest moving this in the else branch of `if (!input_string_size)`.

Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-regexp-object.cpp]: obviously :) It would look better, so I'll do it. :)

**Comment by LaszloLango** [comment]
@akiss77, I've updeted the patch. Please check.

**Comment by akiss77** [comment]
LGTM (informally)

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**